File: FCCU/Theory of Prog/Assigment 1/bubble_sort.js
function bubbleSort(arr) {
    let n = arr.length;
    for(let i = 0; i < n - 1; i++) {
        for(let j = 0; j < n - i - 1; j++) {
            if(arr[j] > arr[j+1]) {
                [arr[j], arr[j+1]] = [arr[j+1], arr[j]];
            }
        }
    }
    return arr;
}

let arr = [5, 3, 8, 4, 2];
console.log(bubbleSort(arr));

File: FCCU/Theory of Prog/Assigment 1/bubble_sort.rs
fn bubble_sort(arr: &mut Vec<i32>) {
    let n = arr.len();
    for _ in 0..n {
        for j in 0..n - 1 {
            if arr[j] > arr[j + 1] {
                arr.swap(j, j + 1);
            }
        }
    }
}

fn main() {
    let mut arr = vec![5, 3, 8, 4, 2];
    bubble_sort(&mut arr);
    println!("{:?}", arr);
}
